resolwe/flow/models.py

```diff
@@ -361,8 +361,10 @@ class Meta(BaseModel.Meta):
     descriptor = JSONField(default={})
 
     def save(self, *args, **kwargs):
+        force_render_descriptor = kwargs.pop('force_render_descriptor', False)
+        defer_render_descriptor = kwargs.pop('defer_render_descriptor', False)
 
-        if not self.pk:  # create
+        if force_render_descriptor or (not defer_render_descriptor and not self.pk):  # create
```

Comment:
Oh, so you mean the second code block besides `render_descriptor`? I am not sure, so in your commit, you added this whole section, so I thought that this was somehow related and put a block around everything. If this is not the case, then the block below should be excluded.